### PR TITLE
feat: quests page

### PR DESCRIPTION
**Pull request by VanishMax**

### CODE REVIEW

**Review by vacekj — APPROVED** [review]

**Comment by hdevalence** [comment]
Some comments / questions (focusing here on the technical parts, we can refine help text later):

1. Is it easy or difficult to reuse UI components to show balances on the "Viewing Balances" page? It could be a good way to show that connected sites can query your data

Screenshot 2024-10-10 at 12 25 33 PM

2. I made an IBC transfer of 1 USDC through Skip and observed it land on Penumbra but it doesn't show up on the IBC page. I'm not sure the past transfers are correct, either, I deposited ~200UM from Osmosis in this wallet but it's listed as "from Penumbra" and the [link it generates](https://ibc.range.org/ibc/status?id=penumbra-1/channel-4/11844) doesn't work: 

Screenshot 2024-10-10 at 12 27 27 PM

3. This should probably be out of scope for the current PR, but in the future we could have a quest of sending funds to another of the user's accounts and use that to explain accounts, memos, etc

Aside from the IBC transfer recognition I think this is good to merge and then iterate on

**Comment by vacekj** [comment]
Diagnosing issue #2, 1. is easy and will be added, 3. - agree, though for after this PR gets merged

**Comment by vacekj** [comment]
@hdevalence implemented number 1 and found and fixed issue with number 2. Show old deposits is now a checkbox. By default, the page will only display deposits since the page was opened.

Tested working on my end, would love confirmation from your end.

File: components/Balances.tsx
Comment (JasonMHasperhoven) [inline components/Balances.tsx:12]: Not a big deal, but you can avoid the first loop by doing:
```suggestion
        .map(({ balanceView }) => (
```

File: components/Deposit.tsx
Comment (JasonMHasperhoven) [inline components/Deposit.tsx:40]: You left a console log here

File: components/Deposit.tsx
Comment (JasonMHasperhoven) [inline components/Deposit.tsx:52]: And here as well

File: components/Deposit.tsx
Comment (JasonMHasperhoven) [inline components/Deposit.tsx:99]: Left commented code here.

File: components/Staking.tsx
Comment (JasonMHasperhoven) [inline components/Staking.tsx:53]: One more console log

**Review by JasonMHasperhoven — APPROVED** [review]
Only some minor non-blocking comments, rest LGTM.

File: pages/web/prax.mdx
Comment (TalDerei) [inline pages/web/prax.mdx:12]: comment: based on slack discussion, filed https://github.com/prax-wallet/prax/issues/218

File: components/hooks.ts
Comment (TalDerei) [inline components/hooks.ts:68]: suggestion: could we also add a note about refreshing the screen?

https://github.com/user-attachments/assets/1d63b18c-ae8f-48a9-9a12-ba7db411a7e1

File: pages/web/prax.mdx
Comment (TalDerei) [inline pages/web/prax.mdx:3]: suggestion: on the top-level [web](https://guide.penumbra.zone/web) page, it could be helpful to mention that this is an interactive, step-by-step guide intended to be followed in order.

This relates to my Slack comment about the loading indicators on the "Shield Funds," "Swapping Tokens," "Staking," and "Disconnect" pages showing indefinite loading. I intentionally skipped steps to see what would happen, and adding a **note** about this might help prevent confusion.

File: components/WalletInstall.tsx
Comment (TalDerei) [inline components/WalletInstall.tsx:75]: uncommented code

File: components/Deposit.tsx
Comment (TalDerei) [inline components/Deposit.tsx:28]: suggestion: `useNotes()` and `useSwaps()` hooks lack a fallback mechanism if the request fails. It'd be prudent to handle the error state for react query:

```
const { data, error, isLoading } = useNotes();
// display error message {error}
```

File: components/Staking.tsx
Comment (TalDerei) [inline components/Staking.tsx:177]: suggestion: create an issue to address the use of `@ts-expect-error` in multiple places which indicates unresolved type issues.

**Review by TalDerei — APPROVED** [review]
left some minor comments, LGTM!

File: pages/web/prax.mdx
Comment (TalDerei) [reply pages/web/prax.mdx:3]: maybe something like

`"Welcome to the interactive guide for the Prax Wallet web extension. This step-by-step tutorial is designed to help you navigate and use the Prax Wallet effectively. Please follow each section in order to ensure the best experience."`

**Comment by TalDerei** [comment]
@vacekj I find some of the information contained within the `div` tags overly verbose. For instance, [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-5c82aee18c205d984ad274909082d63af067bd8349a50c384c00d0687b4a5c97R51-R62) could be improved by breaking down longer sentences for better readability and using bullet points for concepts like the passphrase and recovery phrase. 

another example is [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-c36ceddd1a2b093f2e606e647322bcf63ef63ab19b2a9d86461ab39ebc5b27f5R11-R15), which could be simplified by breaking it into four distinct steps in a numbered list.

Similarly, [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-dc0180fcea0b5682da7bdb9188442923aeccebf2fd3e308d58162b16b434494cR5-R8) and [this one](https://github.com/penumbra-zone/guide/pull/30/files#diff-7bacdd7adf1aed1f74b4719233764c71c879b0e51c913b6a48b5aa65fad90573R5-R8) would also benefit from using numbered steps to make it more digestible.

**Comment by conorsch** [comment]
Big PR, great work! We've got three approvals now, so I'm going to merge. @VanishMax consider filing follow-up issues for any further action related to the discussion above.

File: pages/web/prax.mdx
Comment (vacekj) [reply pages/web/prax.mdx:12]: Thanks!

File: pages/web/prax.mdx
Comment (vacekj) [reply pages/web/prax.mdx:3]: Agree, adding a note to the top-level page.

File: components/Staking.tsx
Comment (vacekj) [reply components/Staking.tsx:177]: Addressing this in https://github.com/orgs/prax-wallet/projects/1/views/7?query=sort%3Aupdated-desc+is%3Aopen&pane=issue&itemId=84056389&issue=penumbra-zone%7Cguide%7C38

File: components/Deposit.tsx
Comment (vacekj) [reply components/Deposit.tsx:28]: Yes, I plan on detecting whether wallet is connected, since that's realistically the only thing besides a connection error that would cause this to fail.

**Comment by vacekj** [comment]
> @vacekj I find some of the information contained within the `div` tags overly verbose. For instance, [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-5c82aee18c205d984ad274909082d63af067bd8349a50c384c00d0687b4a5c97R51-R62) could be improved by breaking down longer sentences for better readability and using bullet points for concepts like the passphrase and recovery phrase.
> 
> another example is [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-c36ceddd1a2b093f2e606e647322bcf63ef63ab19b2a9d86461ab39ebc5b27f5R11-R15), which could be simplified by breaking it into four distinct steps in a numbered list.
> 
> Similarly, [this section](https://github.com/penumbra-zone/guide/pull/30/files#diff-dc0180fcea0b5682da7bdb9188442923aeccebf2fd3e308d58162b16b434494cR5-R8) and [this one](https://github.com/penumbra-zone/guide/pull/30/files#diff-7bacdd7adf1aed1f74b4719233764c71c879b0e51c913b6a48b5aa65fad90573R5-R8) would also benefit from using numbered steps to make it more digestible.

agree, copywriter will handle this.

**Pull request merged**
